File: java/eclipse-workspace/anclient.jserv/src/io/odysz/jclient/tier/ErrorCtx.java
package io.odysz.jclient.tier;

import io.odysz.semantic.jprotocol.AnsonMsg.MsgCode;
import io.odysz.semantic.jprotocol.JProtocol.OnError;

/**This type is used for keeping error handling consist with React TS version, 
 * which is the error handler provided by React context provide. It's the optimized
 * way of abstracting error handling.
 * 
 * @author ody
 *
 */
public class ErrorCtx implements OnError {
    public String msg = "";
	public MsgCode code = null;

	public String msg () { return msg; };

	@Override
	public void err(MsgCode code, String msg, String ... args) {
		this.code = code;
		this.msg = String.format(msg, args);
	}

	public ErrorCtx setSignal(String signal) { return this; };
	public ErrorCtx notifySignal() { return this; };
}

File: csharp/anclient/anclient.net/src/jserv/tier/ErrorCtx.cs

using static io.odysz.semantic.jprotocol.AnsonMsg;

namespace anclient.net.jserv.tier
{
    public abstract class ErrorCtx
    {
        public string msg() { return ""; }

        virtual public void onError(MsgCode code, string msg, string[] args = null)
        {
        }

        public ErrorCtx setSignal(string signal) { return this; }
        public ErrorCtx notifySignal() { return this; }
    }
}
